diff -Nur e2salvage-0.0.8a.orig/src/main.c e2salvage-0.0.8a/src/main.c
--- e2salvage-0.0.8a.orig/src/main.c	2003-10-07 20:49:37.000000000 +0000
+++ e2salvage-0.0.8a/src/main.c	2004-03-13 20:56:16.000000000 +0000
@@ -42,7 +42,7 @@
 	long sb_auto_find = -1;
 	
 	struct hd_geometry geo;
-	struct hd_big_geometry geob;
+	struct hd_geometry geob;
 	struct stat st;
 	unsigned long tmp;
 
diff -Nur e2salvage-0.0.8a.orig/src/main.h e2salvage-0.0.8a/src/main.h
--- e2salvage-0.0.8a.orig/src/main.h	2003-10-07 20:49:37.000000000 +0000
+++ e2salvage-0.0.8a/src/main.h	2004-03-13 20:54:47.000000000 +0000
@@ -2,7 +2,10 @@
 #define _MAIN_H
 
 #include <linux/fs.h>
-#include <sys/types.h>
+#include <linux/types.h>
+
+
+#define HDIO_GETGEO_BIG                0x0330
 
 extern unsigned long blocksize;
 extern unsigned long nr_blocks;
